rmf_fleet_adapter/src/rmf_fleet_adapter/agv/RobotContext.hpp

```diff
@@ -112,6 +116,30 @@ class RobotContext
       const TableViewerPtr& table_viewer,
       const ResponderPtr& responder) final;
 
+  /// Set the state of this robot at the end of its current task
+  RobotContext& current_task_end_state(const rmf_task::agv::State& state);
+
+  /// Get a mutable reference to the state of this robot at the end of its
+  // current task
+  rmf_task::agv::State& current_task_end_state();
```

Comment:
It seems odd to provide this mutable access and also provide a setter function. This makes it impossible for a `const RobotContext&` to view this State field. I would recommend changing this function signature to

```
const rmf_task::agv::State& current_task_end_state() const;
```

Reply:
Fixed https://github.com/osrf/rmf_core/pull/217/commits/51031ae5fcf45f5074c35aecdbb8df5ccc6f9976